### PR TITLE
NewVerticalDataLoader

### LINKED ISSUE #3: Simple Vertically Partitioned Model
## Description
Implement a simple SplitNN using `syft` which trains on vertically split data provided by multiple data holders

## Why?
This may be the first open source implementation of a split model trained on vertically partitioned data. Training the model successfully is a complex process, which requires a custom data pipeline and model architecture.

This epic is the first stage of the project: The data will be ordered (no need for PSI) and we will send data labels away from the data holders

## Breakdown
Provide a bulleted or numbered list of how you might break this epic down into smaller issues.

- [x] Implement function to vertically split dataset (2 data holders/ 1 split) #4 
- [ ] Implement vertically split data loader #5 
- [x] Develop vertically split SplitNN #6 

## Who else?
May require changes in [PySyft](https://github.com/OpenMined/PySyft)

### LINKED ISSUE #5: Vertically Partitioned Data Loader
## Feature Description
Develop a torch dataloader which holds two vertically partitioned sets of the same data.
This dataloader must send the parts of the data to the correct location of their corresponding network

## Additional Context
Part of EPIC #3
Dependent on #6

### PR DESCRIPTION
**Pull request by pavlos-p**
## Description
The `NewVerticalDataLoader` class gets the dataset as input, creates 2 vertically-partitioned datasets, and returns them.

For now, it uses hardcoded `partition_dataset(dataset, remove_data=False, keep_order=True)`, in a future PR, it could take the `remove_data` and `keep_order` as arguments.

_I have tried to use the existing `VerticalDataLoader` and `PartitionDistributingDataLoader` classes in `dataloader.py` but I was getting errors, so I have just created a new class._

Created a `len` method on the `NewVerticalDataLoader` class, instead of using a hardcoded length as it was in the "Simple Vertically Partitioned SplitNN" Jupyter Notebook example.

Closes #5 
Closes #3 

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: src/dataloader.py [hunk 1]
```diff
@@ -59,3 +63,29 @@ def __init__(self, dataset1, dataset2, *args, **kwargs):
 
     def __iter__(self):
         return zip(self.dataloader1, self.dataloader2)
+
+
+class NewVerticalDataLoader:
```
Comment (TTitcombe) [inline src/dataloader.py:68]: With this dataloader, I think we can remove `PartitionDistributingDataLoader`.

To clarify roles, I would rename the new dataloader to `VerticalDataLoader` and rename the existing `VerticalDataLoader` to `SinglePartitionDataLoader`.

Comment (pavlos-p) [reply src/dataloader.py:68]: I agree. Maybe we should Merge the PR for now to avoid more conflicts with the PSI PR #23? And then change these in a future PR?

Comment (TTitcombe) [reply src/dataloader.py:68]: Okay
